betty/deriver.py

```diff
@@ -190,19 +213,30 @@ def _get_derivable_events(person: Person, derivable_event_type: type[EventType])
         yield event
 
 
-def _get_reference_events(person: Person, reference_event_types: set[type[EventType]]) -> Iterable[Event]:
+def _get_reference_events(
+    person: Person,
+    reference_event_types: set[type[EventType]],
+    derivable_event_type: type[EventType],
+) -> Iterable[Event]:
     for presence in person.presences:
         reference_event = presence.event
 
         if reference_event is None:
             continue
 
-        # We cannot reliably determine dates based on reference events with calculated date ranges, as those events
-        # would start or end *sometime* during the date range, but to derive dates we need reference events' exact
-        # start and end dates.
-        if isinstance(reference_event, DerivedEvent):
+        if reference_event.date is None:

```

Comment:
We do nees to take fuzzy dates into account. For example, we must not derive a death that comes before a lifetime event.